### PR TITLE
Feature/cc 0060 add preview and remove saved design in folder list

### PR DESCRIPTION
**Pull request by Evezzm**
Add below information showing in the folder list:
Design preview
Design name
Court type
Saving date
Delete icon

image
image

### CODE REVIEW

**Comment by tatoMa** [comment]
Maybe add hover state of the delete icon button?
image

File: src/components/FolderList/FolderListItems.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        <Tag
+          backgroundColor="tag.courtCategory"
+          size={"sm"}
+          style={{
+            borderRadius: "0",
+            fontSize: "9px",
+            marginBottom: "5px",
+            fontWeight: "bold",
+            textAlign: "center",
+            padding: "0 15px 0 15px",
```
Comment (tatoMa) [inline src/components/FolderList/FolderListItems.tsx:87]: For padding, "0 15px" is equal to "0 15 0 15", When only 2 numbers are in padding, it means X and Y. When 4 numbers are in it means top, right, bottom, left.

Btw: Using numbers that can multiply by 8 or 16 will be better. Normally as frontend dev, we don't use something like 15px or 9px. Cos the default browser rem is 16px.

File: src/interfaces/design.d.ts [hunk 2]
```diff
@@ -6,6 +6,14 @@ export interface IDesign {
   courtSize: ICourtSize;
 }
 
+export interface IDesignDetailed extends IDesign {
```
Comment (zhuocun) [inline src/interfaces/design.d.ts:9]: Is "detailed" should be "detail"? :)

Comment (Evezzm) [reply src/interfaces/design.d.ts]: Thanks Tim! "detail" sounds make more sense - have already renamed it :)

File: src/components/FolderList/FolderListItems.tsx (hunk 1 above)
Comment (Evezzm) [reply src/components/FolderList/FolderListItems.tsx]: Thanks Tao总!  Have updated the format of padding and the numbers in 8x. I learned a lot :)

File: src/components/EditorSideBar/Folder.tsx [hunk 3]
```diff
@@ -7,17 +7,22 @@ import { mockTileData } from "../MockData/MockTileData";
 import { useStoreSelector } from "@/store/hooks";
 import { changeWholeCourtColor } from "@/store/reducer/tileSlice";
 import { ActionCreators } from "redux-undo";
+import FolderListItem from "../FolderList/FolderListItems";
 
-const Folder: React.FC = () => {
+const Folder = () => {
   const dispatch = useDispatch();
   const { designsData, activeCourt } = useStoreSelector((state) => state.courtSpecData);
+  console.log(designsData);
```
Comment (zhuocun) [inline src/components/EditorSideBar/Folder.tsx:15]: Forget to delete them, right? :)

File: src/components/EditorSideBar/Folder.tsx [hunk 4]
```diff
@@ -7,17 +7,22 @@ import { mockTileData } from "../MockData/MockTileData";
...
+  console.log("below is active");
+  console.log(activeCourt);
   const [activateDesign, setActivateDesign] = useState<string>(activeCourt.courtId);
   const { designTileList } = useStoreSelector((state) => state.designTileList);
 
-  useEffect(() => {
-    if (designsData === undefined) return;
-    setActivateDesign(activeCourt.courtId);
-  }, [designsData]);
+  // useEffect(() => {
```
Comment (zhuocun) [inline src/components/EditorSideBar/Folder.tsx:21]: Please remove this function if it is useless :)

Comment (Evezzm) [reply src/components/EditorSideBar/Folder.tsx:21]: Thanks Tim for picking this up~ it's removed now.

**Review by zhuocun — APPROVED** [review]
LGTM

**Review by tatoMa — APPROVED** [review]
LGTM

**Pull request merged**